RDKEMW-5238: Configure persistent path in LISA plugin and upstream patch

jithajamesp opened the pull request:
Reason for change: Realtek VA requires the installation path for DAC Apps to be /tmp/data so adding variable that can be configured from product layer. Changes to upstream https://github.com/rdkcentral/meta-rdk-video/blob/develop/recipes-extended/entservices/entservices-lisa/0100-update-config.patch is also added

Test Procedure: Build and verify.

Risks: Low

jithajamesp committed:
RDKEMW-5238: Configure persistent path in LISA plugin and upstream patch

Reason for change: Realtek VA requires the installation path for DAC Apps to be /tmp/data so adding variable that can be configured from product layer. Changes to upstream https://github.com/rdkcentral/meta-rdk-video/blob/develop/recipes-extended/entservices/entservices-lisa/0100-update-config.patch is also added

Test Procedure: Build and verify.

Risks: Low

Signed-off-by: Jitha James <jitha_james@comcast.com>

jithajamesp commented on LISA/LISA.conf.in:
```diff
@@ -1,12 +1,12 @@
-configuration.add("appspath", "/mnt/apps/dac/images")
-configuration.add("datapath", "/mnt/data/dac")
-configuration.add("annotationsFile", "annotations.json")
+if not boolean("@PLUGIN_LISA_IMAGE_PATH@"):
```
bcatrysse 3 hours ago
I guess this refers to https://github.com/rdkcentral/meta-rdk-video/pull/662/files#diff-5c24d36c3af80ca70a684c65b6b2230f6cbe28b963131ce911c3853f61332fd4R20 ?
and now it will refer to LISA_DEFAULT_PATH ?= "/opt" as default
You would need 2 different paths, ones for dac images storage and one for persistent storage that each app (data path)
Then I would say default should be LISA_DEFAULT_IMAGE_STORAGE_PATH ?= "/opt/dac_apps"
Then I would say default should be LISA_DEFAULT_APPDATA_STORAGE_PATH ?= "/opt/dac_apps/data"

and configuration.add("dbpath", "@PLUGIN_LISA_IMAGE_PATH@/db")
and configuration.add("appspath", "@PLUGIN_LISA_IMAGE_PATH@/apps")
and configuration.add("datapath", "@PLUGIN_LISA_APPDATA_PATH@")

so also needs associated update of the recipe

jithajamesp replied:
Change added.

jithajamesp commented on LISA/LISA.conf.in:
```diff
@@ -1,12 +1,12 @@
+	configuration.add("appspath", "@PLUGIN_LISA_IMAGE_PATH@/apps")
+if not boolean("@PLUGIN_LISA_APPDATA_PATH@"):
+	configuration.add("datapath", "@PLUGIN_LISA_APPDATA_PATH@")
+configuration.add("annotationsFile", "config.json")
```
bcatrysse 3 hours ago

Why do want to change file of annotations.json file, this will break functionality for some bundles. On some bundles there is extra file (next to config.json) for which convention is to name it annotations.json file. So if you will change that filename in config it won't work anymore to get content from that specific annotation file out of it anymore

jithajamesp replied:
This change is part of existing patch: https://github.com/rdkcentral/meta-rdk-video/blob/develop/recipes-extended/entservices/entservices-lisa/0100-update-config.patch#L45

jithajamesp commented on LISA/LISA.conf.in:
```diff
@@ -1,12 +1,12 @@
+configuration.add("annotationsFile", "config.json")
 configuration.add("annotationsRegex", "public\\.*")
-rootobject = JSON()
-rootobject.add("outofprocess", "true")
```
bcatrysse 3 hours ago

Security wise, doing this by default out of process and not in separate container, is not desired.
But yeah if you want to get rid of patch on rdk side I can live with that https://github.com/rdkcentral/meta-rdk-video/blob/develop/recipes-extended/entservices/entservices-lisa/0100-update-config.patch . Would mean we have to patch this on our side. Today we at Liberty Global run this outofprocess and in separate container

jithajamesp commented:
Hi @bcatrysse 
I have copied your review comments from https://github.com/rdkcentral/LISA/pull/5/files here. I have made the suggested changes.

bcatrysse commented on LISA/LISA.conf.in:
```diff
@@ -1,12 +1,12 @@
-configuration.add("datapath", "/mnt/data/dac")
-configuration.add("annotationsFile", "annotations.json")
+if not boolean("@PLUGIN_LISA_IMAGE_PATH@"):
+	configuration.add("dbpath", "@PLUGIN_LISA_IMAGE_PATH@")
```
I know it was not in patch but adding "db" subdir for storage of database has clear benefits (security, excempt from quota mgnt, dedicated dir vs parent dir), so please change to  "dbpath",@PLUGIN_LISA_IMAGE_PATH@/db"   Aslo, otherwise we as LGI will have to patch our own component again and puts our world again upside down.

jithajamesp replied:
Change added

bcatrysse commented on LISA/LISA.conf.in:
```diff
@@ -1,12 +1,12 @@
+	configuration.add("appspath", "@PLUGIN_LISA_IMAGE_PATH@/apps")
+if not boolean("@PLUGIN_LISA_APPDATA_PATH@"):
+	configuration.add("datapath", "@PLUGIN_LISA_APPDATA_PATH@")
+configuration.add("annotationsFile", "config.json")
```
I find it suspicious that name of "annotation.json" is changed to "config.json" here because there I know that we as LG make usome bundles with that extra "annotation.json" file included (next to config.json) and we required that for well functioning. Maybe for community version there is problem with that but since it was in orginal patch I am ok to alsokeep it here but we need to be mindful about it and check if we should not correct in later change.

jithajamesp replied:
I hope we can proceed without modifications to annotationsFile in this PR. Resolving this comment.

bcatrysse reviewed (changes requested):
one more small change requested
goes together with https://github.com/rdkcentral/meta-rdk-video/pull/690
Requesting review & approval from some else as well from Adam https://github.com/astolcenburg

jithajamesp committed:
RDKEMW-5238: Address review comments for dbpath

Signed-off-by: Jitha James <jitha_james@comcast.com>

bcatrysse reviewed (dismissed):
approving but should only be merged in combination with https://github.com/rdkcentral/meta-rdk-video/pull/690

jithajamesp committed:
RDKEMW-5238: Delete LISA.config file

Reason for change: LISA.config file is used in Thunder2.x to generate LISA.json file. Removing it since it is thunder 4.x in RDKE

Test Procedure: Build and verify.

Risks: Low

Signed-off-by: Jitha James <jitha_james@comcast.com>

jithajamesp commented:
Hi @bcatrysse @astolcenburg , removing LISA.config file since it is used only in thunder 2.x and RDKE uses Thunder 4.x. Hope this change is fine
